src/dune_analytics.py

```diff
@@ -13,8 +17,87 @@
 BASE_URL = "https://dune.xyz"
 GRAPH_URL = 'https://core-hsr.duneanalytics.com/v1/graphql'
 
+
 # --------- Constants --------- #
 
+class DuneParameterType(Enum):
+    """
+    Enum of the 4 distinct dune parameter types
+    """
+    TEXT = 'text'
+    NUMBER = 'number'
+    DATE = 'datetime'
+    LIST = 'list'
+
+
+class QueryParameter:
+    """Class whose instances are Dune Compatible Query Parameters"""
+
+    def __init__(
+            self,
+            name: str,
+            parameter_type: DuneParameterType,
+            raw_value: Any,
+            value_str: str
```

Comment:
I think `value_str` can be dropped. It's only needed in `to_dict` and this function should have all the information it needs to correctly serialize `raw_value`.

Reply:
The problem is knowing what to serialize as (I thought it would be good to keep that in the initialization method and not have to serialize more than once. I can also move it into the `to_dict` method if you insist.

Reply:
Seems like a code smell to have basically the same data duplicated.
One could have a switch over the `parameter_type` to know how to deserialize the value.
Alternatively one could also drop `raw_value`. So far it only gets used in `value()` and to be honest I don't expect that function to be of much use.

Reply:
Yea cool. So I will rework this right MEOW and you can let me know what you think. I totally agree that its a bit weird.

Reply:
gefixed!